### PR TITLE
added store class

### PR DESCRIPTION
**Pull request by Kuba-Wi**
Task 1 from second homework.
Also added some methods in player class to make buying and selling work correctly.

### CODE REVIEW

File: shm/include/player.hpp [hunk 1]
```diff
@@ -7,11 +8,16 @@ class Player {
...
-    size_t getAvailableSpace_() const { return availableSpace_; }
+    size_t getAvailableSpace() const { return availableSpace_; }
     size_t getMoney() const { return money_; }
     size_t getSpeed() const { return ship_->getSpeed(); }
     Cargo* getCargo(size_t index) const;
 
+    void setMoney(size_t money) { money_ = money; }
+
+    void loadShip(std::shared_ptr<Cargo> cargo);
+    void unloadShip(Cargo* cargo);
```
Comment (WojtekMs) [inline shm/include/player.hpp:19]: Why do you use std::shared_ptr in loadShip argument and raw ptr in unloadShip?

File: shm/include/store.hpp [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    Store(const std::vector<std::shared_ptr<Cargo>>& stock);
+
+    Response buy(Cargo* cargo, Player* player);
+    Response sell(Cargo* cargo, Player* player);
+
+private:
+    std::vector<std::shared_ptr<Cargo>> stock_;
+
+    void load(std::shared_ptr<Cargo> cargo);
+    void unload(Cargo* cargo);
```
Comment (WojtekMs) [inline shm/include/store.hpp:22]: The same question as in Player class - why would you use shared_ptr in load function argument and raw ptr in unload func?

**Review by WojtekMs — APPROVED** [review]
Great work

File: shm/include/player.hpp (hunk 1 above)
Comment (Kuba-Wi) [reply shm/include/player.hpp:19]: More comfortable would be to use shared_ptr in both places.
In task specification was proposed to use that signatures of functions to learn how to work with APIs that take raw pointers. As far as I know.

File: shm/include/store.hpp (hunk 2 above)
Comment (Kuba-Wi) [reply shm/include/store.hpp:22]: Here I keep the same convenction like above.

**Pull request merged**